app/service/document.py

```diff
@@ -148,16 +148,36 @@ def create(
 
 @db_session.with_transaction(__name__)
 @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
-def delete(import_id: str, context=None, db: Session = db_session.get_db()) -> bool:
+def delete(
+    import_id: str, user_email: str, context=None, db: Session = db_session.get_db()
+) -> Optional[bool]:
     """
     Deletes the document specified by the import_id.
 
     :param str import_id: The import_id of the document to delete.
+    :param str user_email: The email address of the current user.
     :raises RepositoryError: raised on a database error.
     :raises ValidationError: raised should the import_id be invalid.
-    :return bool: True if deleted else False.
+    :return bool: True if deleted None if not.
     """
     id.validate(import_id)
+
+    doc = get(import_id)
+    if doc is None:
+        return None
+
     if context is not None:
         context.error = f"Could not delete document {import_id}"
+
+    entity_org_id = get_org_from_id(db, import_id)
+    app_user.is_authorised_to_make_changes(db, user_email, entity_org_id, import_id)
```

Comment:
Just a note - I wonder if the return value of this function should be removed - and the function renames to something like `raise_if_not_in_organisation` ... just that the function as-is doesn't actually return false so could be misleading - not a blocker - food for thought

Reply:
Yeah I like that idea - I might implement in the latest PR to avoid a bunch of merge conflicts though :)